modules/library/src/main/scala/zio/elasticsearch/executor/response/AggregationResponse.scala

```diff
@@ -159,110 +265,12 @@ private[elasticsearch] final case class FilterAggregationResponse(
-                  sumOfSquares = objFields("sum_of_squares").unsafeAs[Double],
-                  variance = objFields("variance").unsafeAs[Double],
-                  variancePopulation = objFields("variance_population").unsafeAs[Double],
-                  varianceSampling = objFields("variance_sampling").unsafeAs[Double],
-                  stdDeviation = objFields("std_deviation").unsafeAs[Double],
-                  stdDeviationPopulation = objFields("std_deviation_population").unsafeAs[Double],
-                  stdDeviationSampling = objFields("std_deviation_sampling").unsafeAs[Double],
-                  stdDeviationBoundsResponse = objFields("std_deviation_sampling").unsafeAs[StdDeviationBoundsResponse](
-                    StdDeviationBoundsResponse.decoder
-                  )
-                )
-              )
-            case str if str.contains("filter#") =>
-              Some(field -> data.unsafeAs[FilterAggregationResponse](FilterAggregationResponse.decoder))
-            case str if str.contains("max#") =>
-              Some(field -> MaxAggregationResponse(value = objFields("value").unsafeAs[Double]))
-            case str if str.contains("min#") =>
-              Some(field -> MinAggregationResponse(value = objFields("value").unsafeAs[Double]))
-            case str if str.contains("missing#") =>
-              Some(field -> MissingAggregationResponse(docCount = objFields("doc_count").unsafeAs[Int]))
-            case str if str.contains("percentile_ranks#") =>
-              Some(
-                field -> PercentileRanksAggregationResponse(values = objFields("values").unsafeAs[Map[String, Double]])
-              )
-            case str if str.contains("percentiles#") =>
-              Some(field -> PercentilesAggregationResponse(values = objFields("values").unsafeAs[Map[String, Double]]))
-            case str if str.contains("stats#") =>
-              Some(
-                field -> StatsAggregationResponse(
-                  count = objFields("count").unsafeAs[Int],
-                  min = objFields("min").unsafeAs[Double],
-                  max = objFields("max").unsafeAs[Double],
-                  avg = objFields("avg").unsafeAs[Double],
-                  sum = objFields("sum").unsafeAs[Double]
-                )
-              )
-            case str if str.contains("sum#") =>
-              Some(field -> SumAggregationResponse(value = objFields("value").unsafeAs[Double]))
-            case str if str.contains("terms#") =>
-              Some(field -> data.unsafeAs[TermsAggregationResponse](TermsAggregationResponse.decoder))
-            case str if str.contains("value_count#") =>
-              Some(field -> ValueCountAggregationResponse(value = objFields("value").unsafeAs[Int]))
-          }
-      }
-    }.toMap
+    val allFields = BucketDecoder(fields).allFields
+    val docCount  = allFields("doc_count").asInstanceOf[Int]
+    val subAggs   = BucketDecoder(fields).subAggs
```

Comment:
You are creating this two times `BucketDecoder(fields)`. Extract this in val before these three vals.